zerver/lib/markdown/__init__.py

```diff
@@ -64,6 +64,62 @@
 ReturnT = TypeVar("ReturnT")
 
 
+# Taken from
+# https://html.spec.whatwg.org/multipage/system-state.html#safelisted-scheme
+# Must be updated when the list in the HTML spec is updated.
+# Make sure to updated `locless_schemes` as well.
+html_safelisted_schemes = (
+    "bitcoin",
+    "geo",
+    "im",
+    "irc",
+    "ircs",
+    "magnet",
+    "mailto",
+    "matrix",
+    "mms",
+    "news",
+    "nntp",
+    "openpgp4fpr",
+    "sip",
+    "sms",
+    "smsto",
+    "ssh",
+    "tel",
+    "urn",
+    "webcal",
+    "wtai",
+    "xmpp",
+)
+safelisted_schemes = ("http", "https", "ftp", "file") + html_safelisted_schemes
+
+# Checked in https://en.wikipedia.org/wiki/List_of_URI_schemes
+locless_schemes = (
+    "file",
+    "bitcoin",
+    "geo",
+    "im",
+    #"irc",
+    #"ircs",
+    "magnet",
+    "mailto",
+    "matrix",  # TODO can't find an example in the Wikipedia page, so can't confirm.
```

Comment:
I'm not sure if this is matrix.org or something else.

Reply:
Confirmed it is matrix.org: https://matrix.org/blog/2021/04/23/this-week-in-matrix-2021-04-23#matrix-uri-scheme.